src/daemon/krillserver.rs

```diff
@@ -294,8 +171,78 @@ impl KrillServer {
+        let testbed_handle = testbed_ca_handle();
+
+        if let Some(testbed) = config.testbed() {
+            if !server.ca_manager.has_ca(&testbed_handle)? {
+                let testbed_ca = api::import::ImportCa::new(
+                    testbed_handle,
+                    vec![api::import::ImportParent::new(
+                        ta_handle().into_converted(),
+                        ResourceSet::all(),
+                    )],
+                    vec![],
+                );
+
+                let mut import_cas = vec![testbed_ca];
+
+                if let Some(benchmark) = &config.benchmark {
+                    info!("Enabling BENCHMARK mode - ONLY USE THIS FOR TESTING!");
+                    info!(
+                        "Will use import function to create testbed CA and {} CAs with {} ROAs each.",
+                        benchmark.cas, benchmark.ca_roas
+                    );
+
+                    let testbed_parent: ParentHandle = testbed_ca_handle().into_converted();
+                    for nr in 0..benchmark.cas {
+                        let handle = CaHandle::new(format!("benchmark-{}", nr).into());
+
+                        // derive resources for benchmark ca
+                        let byte_2_ipv4 = nr / 256;
+                        let byte_3_ipv4 = nr % 256;
+
+                        let prefix_str = format!("10.{}.{}.0/24", byte_2_ipv4, byte_3_ipv4);
+                        let resources = ResourceSet::from_strs("", &prefix_str, "")
+                            .map_err(|e| Error::ResourceSetError(format!("cannot parse resources: {}", e)))?;
+
+                        // Create ROA configs
+                        let mut roas: Vec<RoaConfiguration> = vec![];
+                        let asn_range_start = 64512;
+                        for asn in asn_range_start..asn_range_start + benchmark.ca_roas {
+                            let payload = RoaPayload::from_str(&format!("{} => {}", prefix_str, asn)).unwrap();
+                            roas.push(payload.into());
+                        }
+
+                        import_cas.push(api::import::ImportCa::new(
+                            handle,
+                            vec![api::import::ImportParent::new(testbed_parent.clone(), resources)],
+                            roas,
+                        ))
+                    }
+                } else {
+                    info!("Enabling TESTBED mode - ONLY USE THIS FOR TESTING AND TRAINING!");
+                    info!("Will use import function to create testbed CA.");
+                }
+
+                let startup_structure = api::import::Structure::new(
+                    testbed.ta_aia().clone(),
+                    testbed.ta_uri().clone(),
+                    testbed.publication_server_uris(),
+                    import_cas,
+                );
+                server.cas_import(startup_structure).await?;
+            } else if config.benchmark.is_some() {
```

Comment:
I'd find it more readable if these two short if blocks came before the longer block.

Reply:
improved (I think)